### PR TITLE
Feature/v2 rads adt

### LINKED ISSUE #660: Upgrade Python converter for rads_adt to IODA-v2:
Update the converter identified in the title rads_adt2ioda to use the IODA-v2 API.

- Update import statement
- Variable dimensions dictionary
- Dimensions dictionary
- Global attributes dictionary
- Remove record numbers
- Update keyDict
- Units, Fill values, other metadata
- Change order of when writer is called
- Remove ExtractoObsData if possible
- Update CMakeLists.txt with environment variable
- Run test to make sure it runs (but will fail because of unupdated test reference)
- Update test reference file
- run unit tests making use of the target converter and verify that they pass. Make sure you verify the results by examining the file using ```ncdump```, ```h5dump```, ```HDFview```, or a similar utility.

## ZenHub organization

Please make sure that you have properly set:
- Labels: OBS3
- Milestone: OBS - October 2021
- Estimate: An estimate of how long this issue will take to resolve. 1 = half a day. 2 = full day.
- Epics: Modify converters to IODAv2 format
- Assignees: Anyone **working** on this issue. Anyone who is not working directly on the issue, but
  who should receive notifications, should be listed below.
- Reviewers: 

## Notifications

Fill in a list of GitHub handles here. People here are not *assigned*, but instead receive notifications
about this issue's progress.

Notifications should always go to: @rhoneyager.

Other people:

### PR DESCRIPTION
**Pull request by kbhargava**
## Description

Upgrades the marine/rads_adt2ioda convert to use the new ioda v-2 interface. 

### Issue(s) addressed

- fixes #660

### CODE REVIEW

File: src/marine/rads_adt2ioda.py [hunk 1]
```diff
@@ -41,37 +60,28 @@ def _read(self):
...
 
-        valKey = vName, self.writer.OvalName()
-        errKey = vName, self.writer.OerrName()
-        qcKey = vName, self.writer.OqcName()
+        valKey = vName, iconv.OvalName()
+        errKey = vName, iconv.OerrName()
+        qcKey = vName, iconv.OqcName()
+        self.VarAttrs[ vName, iconv.OvalName()]['_FillValue'] = Fill_val
+        self.VarAttrs[ vName, iconv.OvalName()]['units'] = val_units
+        self.VarAttrs[ vName, iconv.OvalName()]['scale_factor'] = scale_factor
```
Comment (srherbener) [inline src/marine/rads_adt2ioda.py:75]: Please add attributes "_FillValue" and "units" for the OerrName and OqcName versions of the variable.
* _FillValue for OerrName and OqcName variables should match what the data provider supplies (which are not necessarily the same as the OvalName variable, but typically are the same).
* Units for the OerrName variable can have the value from val_units (ie same as OvalName variable).
* Units for the OqcName variable should have the value 'unitless'.

File: src/marine/rads_adt2ioda.py [hunk 2]
```diff
@@ -20,18 +20,36 @@
...
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string")
+]
+
+GlobalAttrs = {
+}
+
+VarDims = {
+    'adt': ['nlocs'],
```
Comment (CoryMartin-NOAA) [inline src/marine/rads_adt2ioda.py:38]: This is intended to be the `vName` I think, `adt` is not used anywhere is it? However, it is not coded correctly at the moment and thus is ignored. But will be fixed in my GSI-ncdiag PR.

Comment (kbhargava) [reply src/marine/rads_adt2ioda.py]: @CoryMartin-NOAA this has now been taken care of.

**Review by srherbener — APPROVED** [review]
Thanks for addressing my comments. Looks good.

**Review by rhoneyager — APPROVED** [review]

**Pull request merged**